test/integration/patients/worklist/worklist.js

```diff
@@ -3488,67 +3472,78 @@ context('worklist page', function() {
       .find('[data-search-region] .js-input:not([disabled])')
       .as('listSearch')
       .focus()
-      .type('abcd')
-      .then(() => {
-        const storage = JSON.parse(localStorage.getItem(`owned-by_11111_11111-${ STATE_VERSION }`));
-
-        expect(storage.lastSelectedIndex).to.equal(null);
-      });
+      .type('abcd');
 
     cy
       .get('@listSearch')
       .next()
       .click();
 
+    cy
+      .get('@lastTableListItem')
+      .find('.js-select')
+      .click();
+
+    cy
+      .get('.app-frame__content')
+      .find('.table-list__item.is-selected')
+      .should('have.length', 2);
+
     cy
       .get('[data-filters-region]')
       .find('.js-cancel')
       .click();
 
     cy
-      .get('.app-frame__content')
-      .find('.table-list__item')
-      .first()
+      .get('@firstTableListItem')
       .find('.js-select')
       .click();
 
     cy
       .get('.worklist-list__toggle')
-      .contains('Actions')
-      .click()
-      .then(() => {
-        const storage = JSON.parse(localStorage.getItem(`owned-by_11111_11111-${ STATE_VERSION }`));
-
-        expect(storage.lastSelectedIndex).to.equal(null);
-      });
+      .contains('Flows')
+      .click();
 
     cy
       .get('.worklist-list__toggle')
-      .contains('Flows')
+      .contains('Actions')
+      .click();
+
+    cy
+      .get('@lastTableListItem')
+      .find('.js-select')
       .click();
 
+    cy
+      .get('.app-frame__content')
+      .find('.table-list__item.is-selected')
+      .should('have.length', 2);
+
     cy
       .get('[data-filters-region]')
       .find('.js-cancel')
       .click();
 
     cy
-      .get('.app-frame__content')
-      .find('.table-list__item')
-      .first()
+      .get('@firstTableListItem')
       .find('.js-select')
       .click();
 
     cy
       .navigate('/schedule');
 
     cy
-      .go('back')
-      .then(() => {
-        const storage = JSON.parse(localStorage.getItem(`owned-by_11111_11111-${ STATE_VERSION }`));
+      .go('back');
 
-        expect(storage.lastSelectedIndex).to.equal(null);
-      });
+    cy
+      .get('@lastTableListItem')
+      .find('.js-select')
+      .click();
+
+    cy
+      .get('.app-frame__content')
+      .find('.table-list__item.is-selected')
+      .should('have.length', 2);
   });
 
   specify('patient sidebar', function() {
```

Comment:
Same updates as I made for the schedule test.

But there's an additional test to make sure `state.lastSelectedIndex` is cleared when the list is switched between actions and flows (via the `Actions`/`Flows` toggle buttons).